### PR TITLE
Update blocks to Block Protocol 0.3

### PR DESCRIPTION
**Pull request by nathggns**
## 🌟 What is the purpose of this PR?

This PR upgrades most blocks to use 0.3 of the Block Protocol, and to reference production types

- [x] Callout
- [x] Countdown
- [x] Divider
- [x] Heading
- [x] Image
- [x] Paragraph
- [x] Shuffle
- [x] Stopwatch
- [x] Timer
- [x] Video
- [x] Code

Please note, drawing block is updated separately at https://github.com/hashdeps/tldraw-block

## 🔗 Related links

Asana Tasks:

- [Update `stopwatch` block to 0.3](https://app.asana.com/0/1203623179643287/1203921645780991/f)
- [Update `countdown` block to 0.3](https://app.asana.com/0/1203623179643287/1203921490306136/f)
- [Update `divider` block to 0.3](https://app.asana.com/0/1203623179643284/1203921490306137/f)
- [Update `callout` block to 0.3](https://app.asana.com/0/1203623179643287/1203921490306134/f)
- [Update `header` block to 0.3](https://app.asana.com/0/1203623179643287/1203921490306139/f)
- [Update `paragraph` block to 0.3](https://app.asana.com/0/1203623179643287/1203921490306140/f)
- [Update `image` and `video` (media) blocks to 0.3](https://app.asana.com/0/1203623179643281/1203876420195199/f)
- [Update `shuffle` block to 0.3](https://app.asana.com/0/1203623179643287/1203921490306142/f)
- [Update `timer` block to 0.3](https://app.asana.com/0/1203623179643287/1203921645780992/f)

## 🔍 What does this change?

Uses BP 0.3 for all blocks. Removes some functionality from shuffle block to enable this.

### CODE REVIEW

File: blocks/callout/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Entity } from "@blockprotocol/graph";
```
Comment (Alfred-Mountfield) [inline blocks/callout/src/types.ts:1]: [lint](https://github.com/hashintel/hash/pull/2120/commits/59da1019389eefb2099a2095bdd6d17fa9557dee) 

Are these files being linted by eslint?

We might want to disable that, it _does_ say don't edit it 😅  and we probably want to add a CI check later to ensure there isn't a diff

Comment (nathggns) [reply blocks/callout/src/types.ts:1]: Discussed offline – will fix in a follow up https://hashintel.slack.com/archives/C02LG39FJAU/p1677648880232249

File: blocks/image/src/components/media.tsx [hunk 2]
```diff
@@ -348,7 +272,7 @@ export const Media: FunctionComponent<
           onCaptionConfirm={() => updateData({ src: draftSrc })}
           onReset={resetComponent}
           width={draftWidth}
```
Comment (Alfred-Mountfield) [inline blocks/image/src/components/media.tsx:274]: Width is still draggable with `readonly` btw (not sure if it's being persisted)

Comment (nathggns) [reply blocks/image/src/components/media.tsx:274]: Okay – that's an existing flawing with the block's read only implementation. Will fix.

File: blocks/image/src/components/media.tsx [hunk 3]
```diff
@@ -68,112 +65,46 @@ const useDefaultState = <
   return [currentValue, setState];
 };
 
-function getLinkGroup(params: {
```
Comment (Alfred-Mountfield) [inline blocks/image/src/components/media.tsx:71]: This is very satisfying

File: blocks/shuffle/src/shuffle.tsx [hunk 4]
```diff
@@ -186,40 +164,42 @@ export const Shuffle: BlockComponent<BlockEntityProperties> = ({
...
+      outgoingLinks.map((link) => [
+        link.metadata.recordId.entityId,
+        getRightEntityForLinkEntity(
+          blockEntitySubgraph,
+          link.metadata.recordId.entityId,
+        ),
+      ]),
     );
 
     return draftItems.map((item) => {
```
Comment (Alfred-Mountfield) [inline blocks/shuffle/src/shuffle.tsx:188]: Have we removed the ability to add items that represent entities at the moment due to the missing `queryEntityTypes` message?

File: blocks/stopwatch/package.json [hunk 5]
```diff
@@ -15,17 +15,18 @@
     "fix:eslint": "eslint --fix .",
```
Comment (Alfred-Mountfield) [inline blocks/stopwatch/package.json:15]: This isn't on the hub at the moment, we probably don't want to publish it this time around either right?

File: blocks/shuffle/src/shuffle.tsx (hunk 4 above)
Comment (nathggns) [reply blocks/shuffle/src/shuffle.tsx:188]: Yes – you can pass in data that represents that, but you can't load it in via UI. Was agreed on slack somewhere.

File: blocks/stopwatch/src/dev.tsx [hunk 6]
```diff
@@ -2,17 +2,37 @@
  * This is the entry point for developing and debugging.
```
Comment (Alfred-Mountfield) [inline blocks/stopwatch/src/dev.tsx:2]: I'm not seeing the `README.md`s in the diff, have you checked the descriptions are still up to date? I have a feeling they're not, and might be kind of annoying to update 🙃 

They unfortunately appear pretty prominently on the Hub

File: blocks/video/src/components/media.tsx [hunk 7]
```diff
@@ -189,108 +119,86 @@ export const Media: FunctionComponent<
...
+
   const updateData = useCallback(
-    ({ width, src }: { src: string | undefined; width?: number }) => {
-      if (readonly) {
-        return;
-      }
+    ({ src }: { src: string | undefined; }) => {
       if (src?.trim()) {
-        if (graphService?.updateEntity && entityId) {
+        // @todo how to handle this not being defined now
```
Comment (Alfred-Mountfield) [inline blocks/video/src/components/media.tsx:129]: Shall we track this in a follow-up or does it need to be addressed now?

File: blocks/stopwatch/package.json (hunk 5 above)
Comment (CiaranMn) [reply blocks/stopwatch/package.json:15]: Not sure why it wasn't on – might as well publish it if it works

File: blocks/stopwatch/src/dev.tsx (hunk 6 above)
Comment (nathggns) [reply blocks/stopwatch/src/dev.tsx:2]: I have not touched the READMEs at all. Will have a quick look

**Review by Alfred-Mountfield — APPROVED** [review]
Thanks @nathggns it looks great. 

I've tried out all the block's locally and they seem to be working as expected, except for the few known regressions I highlighted.

Quite excited by how clean the code still looks, even with the somewhat hacky type generation, and the `stdlib` _seems_ to be working a treat.

Few outstanding threads, but I defer to your judgement on if they need resolving now or in follow-ups.

This must have been a pain to develop considering all the clashes in ports and such!

File: blocks/video/src/components/media.tsx (hunk 7 above)
Comment (nathggns) [reply blocks/video/src/components/media.tsx]: Comment is out of date, we can just remove the check

File: blocks/image/src/components/media.tsx (hunk 2 above)
Comment (Alfred-Mountfield) [reply blocks/image/src/components/media.tsx:274]: Can be done in follow-up if we like

Comment (nathggns) [reply blocks/image/src/components/media.tsx:274]: It's fixed already – thanks for flagging

File: blocks/stopwatch/src/dev.tsx (hunk 6 above)
Comment (nathggns) [reply blocks/stopwatch/src/dev.tsx:2]: Updated READMEs except for timer which will do tomorrow. Put a note saying it needs updating instead

**Pull request merged**